JANGAN/ProjectTools/BaseKerasModelTrainer.py

```
@@ -0,0 +1,137 @@
+
+            print(f"Time for epoch {epoch + 1} is {self.GetDatetimeFromSeconds(totalEpochTime)}. Est time remaining for training is {self.GetDatetimeFromSeconds(totalEpochTime*(self.Epochs-(epoch + 1)))}")
+        
+        print("Training finished!")
+            
+    def CreateDataSet(self, dataArray):
+        (returnTrainSet, returnTestSet) = dataArray[0]
+        for data in dataArray[1:]:
+            (addTrainSet, addTestSet) = data
+            returnTrainSet = returnTrainSet.concatenate(addTrainSet)
+            returnTestSet = returnTestSet.concatenate(addTestSet)
+        
+        returnTrainSet = returnTrainSet.shuffle(buffer_size=1024)
+        returnTestSet = returnTestSet.shuffle(buffer_size=1024)
+        return (returnTrainSet, returnTestSet)
+
+    def GetDatetimeFromSeconds(self, seconds):
+        return time.strftime("%H:%M:%S", time.gmtime(seconds))
+
+    def PrintStatus(self, iteration, totalIterations, epochTime, epoch):
+        print("No Train loss text implemented.", end="\r")
+
+    def PrintTestStatus(self, iteration, totalIterations, epochTime):
+        print("No Test accuracy text implemented.", end="\r")
+
+    def SaveCheckpoint(self):
+        if os.path.exists(self.CheckpointPath + 'checkpoint.index'):
+            from ProjectTools import HelperFunctions as hf
+            hf.DeleteFolderAndAllContents(self.CheckpointPath)
+
+        ckptPath = self.CheckpointPath + 'checkpoint_' + str(self.CurrentEpoch)
+
+        self.Model.save_weights(ckptPath)
+        self.MakeCheckpointRef(ckptPath, self.LatestCheckpointPath)
+
+    def MakeCheckpointRef(self, ckptPathSrc, ckptPathDest):
+        relPath = os.path.relpath(os.path.abspath(ckptPathSrc), os.path.abspath(ckptPathDest))
+        os.makedirs(os.path.dirname(ckptPathDest), exist_ok=True)
+        with open(ckptPathDest, 'w') as f:
+            f.write(f"{ckptPathSrc}")
+
+    def LogData(self, epoch):
+        print("No Data logging implemented")
```

Is it not better to throw an exception?